src/collection-view.js

```diff
@@ -282,10 +282,11 @@ Marionette.CollectionView = Marionette.AbstractView.extend({
   // Retrieve the `childView` class, either from `this.options.childView`
   // or from the `childView` in the object definition. The "options"
   // takes precedence.
-  // This method receives the model that will be passed to the instance
-  // created from this `childView`. Overriding methods may use the child
-  // to determine what `childView` class to return.
-  getChildView: function(child) {
+  // The `childView` property can be either a view class or a function that
+  // returns a view class. If it is a function, it will receive the model that
+  // will be passed to the view instance (created from the returned view class)
+
+  _getChildView: function(child) {
     var childView = this.getOption('childView');
 
     if (!childView) {
```

Comment:
I don't think this check or error is valuable now.  Its fairly redundant.
`InvalidChildViewError` will be thrown later if there's an issue.